### PR TITLE
fix: convert ProjectIDs type to []string

### PR DESCRIPTION
**Pull request by jing-zhi**
#### What this PR does / why we need it:

fix(performance_measure.go):convert ProjectIDs type to []string]

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](paste your link here)

#### Specified Reviewers:

/assign @chengjoey @sfwn 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |     fix(performance_measure.go):convert ProjectIDs type to []string]         |
| 🇨🇳 中文    |        转换 ProjectIDs 为 []string，类型转换     |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by chengjoey — APPROVED** [review]

File: internal/apps/dop/providers/efficiency_measure/contributor_query.go [hunk 1]
```diff
@@ -158,7 +158,11 @@ func (p *provider) makeContributorBasicSql(req *apistructs.PersonalContributionR
 			Group("repoID").
 			Group("timestamp")
 		if len(req.ProjectIDs) > 0 {
-			tx = tx.Where("tag_values[indexOf(tag_keys,'org_id')] in (?)", req.ProjectIDs)
+			var projectIDs []string
```
Comment (sfwn) [inline internal/apps/dop/providers/efficiency_measure/contributor_query.go:161]: extract a function and add unit-test.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: internal/apps/dop/providers/efficiency_measure/contributor_query.go (hunk 1 above)
Comment (jing-zhi) [reply internal/apps/dop/providers/efficiency_measure/contributor_query.go:161]: ok

File: internal/apps/dop/utils/conversion.go [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import "fmt"
+
+func UintSliceToStringSlice(IDs []uint64) (ids []string) {
+	for _, id := range IDs {
+		ids = append(ids, fmt.Sprintf("%d", id))
+	}
+	return ids
+}
+
+func UintSliceToStringsSlice(IDs []uint64) (ids []string) {
```
Comment (sfwn) [inline internal/apps/dop/utils/conversion.go:26]: function name is not clear.

File: internal/apps/dop/utils/conversion.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import "fmt"
+
+func UintSliceToStringSlice(IDs []uint64) (ids []string) {
```
Comment (sfwn) [inline internal/apps/dop/utils/conversion.go:19]: put these methods into `pkg/strutil`, and use Generics.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: pkg/strutil/convert.go [hunk 4]
```diff
@@ -24,3 +25,19 @@ func NoCopyBytesToString(b []byte) string {
 func NoCopyStringToBytes(s string) []byte {
 	return *(*[]byte)(unsafe.Pointer(&s))
 }
+
+func ToStrSlice[T any](input []T, withQuote ...bool) []string {
```
Comment (sfwn) [inline pkg/strutil/convert.go:29]: add ut here

Comment (sfwn) [reply pkg/strutil/convert.go:29]: test: int, string with single quote, byte, and others

**Review by sfwn — CHANGES_REQUESTED** [review]

File: pkg/strutil/convert.go [hunk 5]
```diff
@@ -24,3 +25,19 @@ func NoCopyBytesToString(b []byte) string {
...
+	result := make([]string, len(input))
+	if len(withQuote) == 0 {
```
Comment (sfwn) [inline pkg/strutil/convert.go:31]: set flag, and iterate just once

**Review by sfwn — CHANGES_REQUESTED** [review]

File: pkg/strutil/convert.go (hunk 5 above)
Comment (jing-zhi) [reply pkg/strutil/convert.go]: ok

File: pkg/strutil/convert.go (hunk 4 above)
Comment (jing-zhi) [reply pkg/strutil/convert.go]: ok

File: internal/apps/dop/utils/conversion.go (hunk 3 above)
Comment (jing-zhi) [reply internal/apps/dop/utils/conversion.go]: ok

File: pkg/strutil/convert.go [hunk 6]
```diff
@@ -24,3 +25,23 @@ func NoCopyBytesToString(b []byte) string {
 func NoCopyStringToBytes(s string) []byte {
 	return *(*[]byte)(unsafe.Pointer(&s))
 }
+
+func ToStrSlice[T uint64 | uint | int64 | int](input []T, withQuote ...bool) []string {
+	result := make([]string, len(input))
+	if len(withQuote) == 0 {
+		withQuote[0] = false
```
Comment (sfwn) [inline pkg/strutil/convert.go:32]: panic here if nothing passed in.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: pkg/strutil/convert.go (hunk 6 above)
Comment (sfwn) [reply pkg/strutil/convert.go:32]: image

**Review by sfwn — APPROVED** [review]

**Pull request merged**